### ISSUE TITLE
Special chars in forum name (:) give gmail grief in mailings

### ISSUE THREAD

**Issue by sbulen** [SimpleMachines/SMF#5834]
#### Description
In this thread m4z confirmed there are problems with mailing when there are special characters (in this case, a colon) in the forum name.  

On the one hand, this seems to me to be a gmail issue.  On the other hand, gmail is mind-bogglingly pervasive.

### Steps to reproduce
1. Put a colon in your forum name
2. At least one recipient should use gmail
3. Send a mailing...

### Environment (complete as necessary)
- Recipient must include gmail...

### Additional information/references
Original thread:
https://www.simplemachines.org/community/index.php?topic=569926.0

**Comment by m4z** [comment]
Interestingly, I just tested sending a newsletter (which also inserts a colon into the subject) and _that_ works fine.

---

And, I meant to comment this here (as I did in the original thread) for weeks:

> On the one hand, this seems to me to be a gmail issue. On the other hand, gmail is mind-bogglingly pervasive.
> 
> ### Steps to reproduce
> 1. Put a colon in your forum name
> 2. At least one recipient should use gmail

I don't think this is only a GMail issue. To summarize:
* If you _don't_ use a colon in the forum title ("`Foo Bar`"), the display name in the `From:` header is fine:
```From: Foo Bar <webmaster@beta-forum.(redacted)>```
* If you do use a colon ("`Foo: Bar`"), stuff breaks:
```From: Foo@mail.(redacted), Bar@mail.(redacted) <webmaster@beta-forum.(redacted)>```


I tested with one throw-away mailer too, and instead of rejecting the mail (like GMail does), it just completely broke displaying the mail in their webmailer.

Also, these words-in-the-forum-title mail addresses (inserted into the display part of the `From:` header by this bug) probably aren't existing (some of them aren't even valid mail adresses in my case).
